src/walk.rs

```diff
@@ -36,13 +36,99 @@ enum ReceiverMode {
 
 /// The Worker threads can result in a valid entry having PathBuf or an error.
 #[allow(clippy::large_enum_variant)]
+#[derive(Debug)]
 pub enum WorkerResult {
     // Errors should be rare, so it's probably better to allow large_enum_variant than
     // to box the Entry variant
     Entry(DirEntry),
     Error(ignore::Error),
 }
 
+/// A batch of items to send over a channel.
+struct Batch<T> {
+    items: Arc<Mutex<Option<Vec<T>>>>,
+}
+
+impl<T> Batch<T> {
+    fn new() -> Self {
+        Self {
+            items: Arc::new(Mutex::new(Some(vec![]))),
```

Comment:
Would it be worth initializing the batch with some capacity?

Reply:
I just did a quick benchmark and it did not make a significant difference.